src/scripts/clipperUI/ratingsHelper.ts

```diff
@@ -0,0 +1,538 @@
+		if (Utils.isNullOrUndefined(badRatingVersion)) {
+			// value has never been set, no bad rating given
+			return true;
+		}
+
+		if (!RatingsHelper.versionHasCorrectFormat(lastSeenVersion) || !RatingsHelper.versionHasCorrectFormat(badRatingVersion)) {
+			return false;
+		}
+
+		let lastSeenMajor: number = RatingsHelper.getMajorVersion(lastSeenVersion);
+		let badRatingMajor: number = RatingsHelper.getMajorVersion(badRatingVersion);
+
+		if (lastSeenMajor > badRatingMajor) {
+			return true;
+		}
+
+		let lastSeenMinor: number = RatingsHelper.getMinorVersion(lastSeenVersion);
+		let badRatingMinor: number = RatingsHelper.getMinorVersion(badRatingVersion);
+
+		if (lastSeenMajor === badRatingMajor && lastSeenMinor > badRatingMinor) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Returns true if ALL of the below applies:
+	 *   * Number of successful clips >= {Constants.Settings.minClipSuccessForRatingsPrompt}
+	 *   * Number of successful clips <= {Constants.Settings.maxClipSuccessForRatingsPrompt}
+	 *   * TODO Number of successful clips is on the "gap between prompts" boundary
+	 *
+	 * Public for testing
+	 */
+	public static clipSuccessDelayIsOver(numClips: number): boolean {
+		// TODO Waiting for consensus on how should we handle users who never interact with the ratings prompt
+			// (keep showing the prompt forever, stop after max # of successful clips, slowly increase "gap between prompts")
+
+		// TODO MVP+? when # of successful clips > nMax, collapse panel into a Rate Us hyperlink in the footer that is always available
+
+		if (isNaN(numClips)) {
+			return false;
+		}
+
+		return numClips >= Constants.Settings.minClipSuccessForRatingsPrompt &&
+			numClips <= Constants.Settings.maxClipSuccessForRatingsPrompt; /*&&
+			numClips % Constants.Settings.gapBetweenClipSuccessForRatingsPrompt === 0;*/
+	}
+
+	/**
+	 * Implementation of the logic described in the shouldShowRatingsPrompt(...) description
+	 */
+	private static shouldShowRatingsPromptInternal(clipperState: ClipperState, event: Log.Event.PromiseEvent, logEventInfo: RatingsLoggingInfo): Promise<boolean> {
+		return new Promise<boolean>((resolve, reject) => {
+			if (Utils.isNullOrUndefined(clipperState)) {
+				event.setStatus(Log.Status.Failed);
+				event.setFailureInfo({ error: "Clipper state is null or undefined" });
+				return reject(undefined);
+			}
+
+			if (!Utils.isNullOrUndefined(clipperState.shouldShowRatingsPrompt)) {
+				// return cached value in clipper state since it already exists
+				logEventInfo.usedCachedValue = true;
+				return resolve(clipperState.shouldShowRatingsPrompt);
+			}
+
+			let ratingsPromptEnabled: boolean = RatingsHelper.ratingsPromptEnabledForClient(clipperState.clientInfo.clipperType);
+			logEventInfo.ratingsPromptEnabledForClient = ratingsPromptEnabled;
+			if (!ratingsPromptEnabled) {
+				return resolve(false);
+			}
+
+			Clipper.Storage.getValue(Constants.StorageKeys.doNotPromptRatings, (doNotPromptRatingsStr) => {
```

Comment:
+1  

There are other places this could be used. For example, getting the cachedNotebooks and curSectionPath in SectionPicker.tsx. I also feel like there are other places in the codebase.